osbuild/util/chroot.py

```diff
@@ -0,0 +1,41 @@
+import os
+import subprocess
+
+
+class Chroot:
+    """
+    Sets up mounts for the virtual filesystems inside a root tree, preparing it for running commands using chroot. This
+    should be used whenever a stage needs to run a command against the root tree but doesn't support a --root option or
+    similar.
+    Cleans up mounts when done.
+
+    This mounts /proc, /dev, and /sys.
+    """
+
+    def __init__(self, root: str):
+        self.root = root
+
+    def __enter__(self):
+        for d in ["/proc", "/dev", "/sys"]:
+            if not os.path.exists(self.root + d):
+                print(f"Making missing chroot directory: {d}")
```

Comment:
My instinct would be to be to just do a `os.makedirs(,exists_okay=True)` here but maybe I'm missing something and the print (and benign TOCTOU) is good to have? Genuinely curious :)

[edit: uh, I see this is just a literal copy/paste, sorry, just ignore me, I can have a look for a followup]